src/Helper/DeclareVars.php

```diff
@@ -45,39 +42,39 @@ class DeclareVars extends AbstractHelper
      * </code>
      *
      * phpcs:ignore
-     * @param string|array variable number of arguments, all string names of variables to test
-     * @return void
+     * @param string|array ...$arguments variable number of arguments, all string names of variables to test
      */
-    public function __invoke()
+    public function __invoke(string|array ...$arguments): void
     {
-        $view = $this->getView();
-        $args = func_get_args();
-        foreach ($args as $key) {
+        foreach ($arguments as $key) {
             if (is_array($key)) {
+                /** @psalm-var mixed $value */
                 foreach ($key as $name => $value) {
+                    if (! is_string($name)) {
+                        continue;
+                    }
+
                     $this->declareVar($name, $value);
                 }
-            } elseif (! isset($view->vars()->$key)) {
-                $this->declareVar($key);
+
+                return;
             }
+
+            $this->declareVar($key);
         }
     }
 
     /**
      * Set a view variable
      *
-     * Checks to see if a $key is set in the view object; if not, sets it to $value.
-     *
-     * @param  string $key
-     * @param  string $value Defaults to an empty string
-     * @return void
+     * Checks to see if a $key is set in the view object; if not, sets it to $value with a default of an empty string.
```

Comment:
WTF? I know this is pre-existing, but how is this supposed to be useful in any way?! 

Let's instead kill the helper for 3.0.x

Reply:
The helper is not documented and does not really seem to be helpful. Remove it!